sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -3769,16 +3769,12 @@ pi_result piextQueueCreateWithNativeHandle(pi_native_handle NativeHandle,
   PI_ASSERT(Context, PI_ERROR_INVALID_CONTEXT);
   PI_ASSERT(NativeHandle, PI_ERROR_INVALID_VALUE);
   PI_ASSERT(Queue, PI_ERROR_INVALID_QUEUE);
+  PI_ASSERT(Device, PI_ERROR_INVALID_DEVICE);
 
   auto ZeQueue = pi_cast<ze_command_queue_handle_t>(NativeHandle);
   // Assume this is the "0" index queue in the compute command-group.
   std::vector<ze_command_queue_handle_t> ZeQueues{ZeQueue};
 
-  // For compatibility with older implementations we allow the device to be
-  // optional for now. Once the deprecated interop API is removed this can be
```

Comment:
Was the no-device API removed already? If not, why not do so in this patch?

Reply:
@steffenlarsen and I discussed before and decided not to remove the legacy make_queue API that does not require 'device' since they are internally used in several places.

Reply:
Not sure I understand the motivation to keep deprecated API instead of rewriting with the new one. Please clarify.
Also the PR title remains not clear: "Make queue2"

Reply:
Sorry, I was talking about something else.
I removed the deprecated make_queue API in backend.hpp now.

Reply:
OK, I see you removed a version of make_queue that didn't have the "KeepOwnership" parameter. Please, make sure there is only single version of "make_queue" remains.

Reply:
That's what I was referring to about Steffen and my decision NOT to remove.
These APIs are actually used in several places in SYCL RT, so we can't remove them.

Reply:
> These APIs are actually used in several places in SYCL RT, so we can't remove them.

Why we can't (or don't want to) rewrite those remaining uses with the new API and remove those deprecated?
@steffenlarsen : can you give your perspective please?

Reply:
Problem is that the previously redundant function is now also used in `sycl/include/sycl/ext/oneapi/experimental/backend/cuda.hpp` where ownership is kept by the user, while in `sycl/include/sycl/backend.hpp` it is used to pass ownership to the backend. If we only had the latter, we could have removed it, but since the ownership argument is used we should keep it.

We should probably entirely remove the comment about removing it given it is actually being used for a valid purpose.

Reply:
Done with removing the comment that is not valid anymore.

Reply:
Please point exactly to what is the "valid" use of an interface that we intended to retire?

Reply:
https://github.com/intel/llvm/blob/sycl/sycl/include/sycl/ext/oneapi/experimental/backend/cuda.hpp#L105 uses it to let the user keep ownership of CUDA native objects, while https://github.com/intel/llvm/blob/sycl/sycl/include/sycl/backend.hpp#L272 uses it to pass ownership to the underlying API for all unspecialized cases.

Reply:
> https://github.com/intel/llvm/blob/sycl/sycl/include/sycl/ext/oneapi/experimental/backend/cuda.hpp#L105 uses it to let the user keep ownership of CUDA native objects, while https://github.com/intel/llvm/blob/sycl/sycl/include/sycl/backend.hpp#L272 uses it to pass ownership to the underlying API for all unspecialized cases.

Fine, but why can't they use the very new interface that is remaining and has the `KeepOwnership`:

``` C++
__SYCL_EXPORT queue make_queue(pi_native_handle NativeHandle,
                               const context &TargetContext,
                               const device &TargetDevice, bool KeepOwnership,
                               const async_handler &Handler, backend Backend);
```

I expect that a single `detail::make_queue` should be sufficient, no?

Reply:
What device would they pass to that function? They could pick one from the context, but in that case we run the risk of the backends disagreeing on the actual associated device. For example, the native queue could be using another device from `TargetContext` than whatever we picked, so is it allowed to ignore the device passed to it?

Reply:
The device is optional, see https://github.com/intel/llvm/blob/a0a4d72155fd6ecc5a77bed3b03eeedef6817358/sycl/include/sycl/detail/pi.h#L1129

Here for CUDA you'd pass nullptr, and ignore it.

Reply:
Right. In that case we need to change the signature of the function you mentioned, as it takes a reference to the device rather than a pointer. I am okay with that solution.

Reply:
Yes, let's avoid the zoo of many similar functions, and just have one.
Note, that we are already passing the "nullptr" in the implementation of the function without the "device":
https://github.com/intel/llvm/blob/a0a4d72155fd6ecc5a77bed3b03eeedef6817358/sycl/source/backend.cpp#L108